apps/src/pythonlab/pyodideWorkerManager.js

```diff
@@ -7,6 +7,7 @@ import {
 } from '@codebridge/redux/consoleRedux';
 
 import {setAndSaveProjectSource} from '@cdo/apps/lab2/redux/lab2ProjectRedux';
+import {setIsRunning} from '@cdo/apps/lab2/redux/systemRedux';
```

Comment:
I was curious why we needed this here and did a little digging. We normally set `isRunning` [here](https://github.com/code-dot-org/code-dot-org/blob/74dd1d1895e22417c305b580fd86c17a4f73e235/apps/src/codebridge/ControlButtons/index.tsx#L117), and I was surprised we weren't hitting that given it's a `finally` block. When I ran a program with an error, there was an error message in the browser console `results.toJs()` is not a function. I realized this was my bad 🤦‍♀️ We are trying to parse the results object [here](https://github.com/code-dot-org/code-dot-org/blob/74dd1d1895e22417c305b580fd86c17a4f73e235/apps/src/pythonlab/pyodideWebWorker.js#L91) but it's initially set to an empty string [here](https://github.com/code-dot-org/code-dot-org/blob/74dd1d1895e22417c305b580fd86c17a4f73e235/apps/src/pythonlab/pyodideWebWorker.js#L64), so if we hit an error earlier we are calling `toJs` on a string, rather than skipping it. If we change the default value to `undefined` then the `finally` block should get hit and we don't need this change.

Reply:
Confirmed that this works! TIL that the `?.` operator only checks if it is `null` or `undefined` (as opposed to generally falsy).

Reply:
Updated branch. Now the program completes as expected and it also logs a message to the student console to show this. A much nicer solution!

<img width="696" alt="image" src="https://github.com/user-attachments/assets/ed301adc-dc5d-4175-a923-d98ca7955220">